resources/iotjs-hardware-tests/tests/test_adc_echo.js

```diff
@@ -0,0 +1,180 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+var assert = require('assert');
+var GPIO = require('gpio');
+var ADC = require('adc');
+//var adc = new ADC();
+var gpio = new GPIO();
+
+var adc0 = null;
+
+var ADC_PIN = 0;
+var GPIO_ADC_MODE_1 = 32;
+var GPIO_ADC_MODE_2 = 30;
+var gpioPins = [];
+var gpioOpenCounter = 0;
+
+function resetGpio() {
+    console.log('adc: reseting gpio');
+    gpioPins.forEach(function (pin, index) {
+        if (pin) {
+            pin.writeSync(0);
+        }
+    });
+}
+
+function endTest() {
+    console.log('adc: closing');
+    adc0.closeSync();
+
+    resetGpio();
+
+    console.log('adc: closing gpio');
+    gpioPins.forEach(function (pin) {
+        if (pin) {
+            pin.closeSync();
+        }
+    });
+}
+
+// we have to use median filtering
+// due tue tizenrt bug which gives
+// first few results wrong
+function medianRead(adcPort, observations, callback, interval) {
+    var results = [],
+        read = function () {
+            setTimeout(function () {
+                adcPort.read(readCallback);
+            }, interval);
+        },
+        readCallback = function (err, value) {
+            var half = 0;
+            if (err) {
+                callback(err);
+            } else {
+                results.push(value);
+            }
+            if (results.length === observations) {
+                results.sort(function (a, b) { return a - b; });
+                half = (results.length / 2) | 0;
+                if (results.length % 2) {
+                    callback(null, results[half]);
+                } else {
+                    callback(null, (results[half - 1] + results[half]) / 2);
+                }
+            } else {
+                read();
+            }
+        }
+
+    if (interval === undefined) {
+        interval = 250; // default read interval in ms
+    }
+
+    if (observations > 0) {
+        read();
+    } else {
+        callback('number of observations not specified');
+    }
+}
+
+function runTest() {
```

Comment:
The McCC complexity of the this function is too big. It would be nice to simplify this function. E.g. using early return statements.

```js
if (err) {
  asert(...);
  return;
}

...

```